From 3ca787d7942c8b00fda25eb47d956a86307a4d9b Mon Sep 17 00:00:00 2001
From: Ovidiu Panait <ovidiu.panait@windriver.com>
Date: Thu, 19 Jul 2018 08:18:56 +0000
Subject: [PATCH] net: ethernet: xilinx: axienet: fix 'lp' undeclared error

Fixes the following build error when CONFIG_XILINX_AXI_EMAC=y and
CONFIG_XILINX_AXI_EMAC_HWTSTAMP is not set:
/drivers/net/ethernet/xilinx/xilinx_axienet_main.c:1729:24: error: 'lp' undeclared (first use in this function)
| return phy_mii_ioctl(lp->phy_dev, rq, cmd);
| ^~

This issue is specific to the WRL9 xilinx-zynq kernel, introduced by
commit d84994f8d27a [drivers: net: ethernet: axienet: fix compilation error].

Signed-off-by: Ovidiu Panait <ovidiu.panait@windriver.com>
---
 drivers/net/ethernet/xilinx/xilinx_axienet_main.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
index 25b7f61..5ecaea8 100644
--- a/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
+++ b/drivers/net/ethernet/xilinx/xilinx_axienet_main.c
@@ -1726,7 +1726,7 @@ static int axienet_ioctl(struct net_device *dev, struct ifreq *rq, int cmd)
 	case SIOCGMIIPHY:
 	case SIOCGMIIREG:
 	case SIOCSMIIREG:
-		return phy_mii_ioctl(lp->phy_dev, rq, cmd);
+		return phy_mii_ioctl(dev->phydev, rq, cmd);
 #ifdef CONFIG_XILINX_AXI_EMAC_HWTSTAMP
 	case SIOCSHWTSTAMP:
 		return axienet_set_ts_config(lp, rq);
-- 
1.7.9.5

